internal/planner/mapper/mapper.go

```diff
@@ -41,7 +41,51 @@ const (
 	CommitSelection
 )
 
-// ToSelect converts the given [parser.Select] into a [Select].
+// ToOperation converts the given [request.OperationDefinition] into an [Operation].
+//
+// In the process of doing so it will construct the document map required to access the data
+// yielded by the [Operation].
+func ToOperation(
+	ctx context.Context,
+	store client.Store,
+	operationRequest *request.OperationDefinition,
+) (*Operation, error) {
+	operation := &Operation{
+		DocumentMapping: core.NewDocumentMapping(),
+	}
+
+	for i, s := range operationRequest.Selections {
+		switch t := s.(type) {
```

Comment:
suggestion: The default case is not handled, it might be better to return an error instead of quietly ignoring it - I believe it could only happen if we mess up in the future, but it could save some head-scratching even if it means a line or two of dead code.

Reply:
I've added a default case with an error.